test/karma.conf.js

```diff
@@ -1,196 +1,41 @@
-      'karma-browserify',
-      'karma-coverage',
-      'karma-detect-browsers',
-    ],
-
-    detectBrowsers: {
-      enabled: false,
-      usePhantomJS: false
-    },
-
-    reporters: ['dots'],
-
-    // web server port
-    port: 9876,
-
-    // cli runner port
-    runnerPort: 9100,
-    colors: true,
-    logLevel: config.LOG_INFO,
-    captureTimeout: 300000,
-    browserNoActivityTimeout: 300000,
-    browserDisconnectTimeout: 300000,
-    browserDisconnectTolerance: 3,
-
-    browserStack: {
-      project: 'Video.js',
-      name: browserstackName,
-      build: browserstackName,
-      pollingTimeout: 30000,
-      captureTimeout: 600,
-      timeout: 600
-    },
-    customLaunchers: getCustomLaunchers(),
+const generate = require('videojs-generate-karma-config');
 
-    // The HTML reporter seems to be busted right now, so we're just using text in the meantime
-    // along with the summary after the test run.
-    coverageReporter: {
-      reporters: [
-        {
-          type: 'text',
-          dir: 'coverage/',
-          file: 'coverage.txt'
-        },
-        {
-          type: 'lcovonly',
-          dir: 'coverage/',
-          subdir: '.'
-        },
-        { type: 'text-summary' }
-      ]
-    },
+module.exports = function(config) {
 
-    // make QUnit show the UI in karma runs
-    client: {
-      clearContext: false,
-      qunit: {
-        showUI: true,
-        testTimeout: 5000
-      }
+  // see https://github.com/videojs/videojs-generate-karma-config
+  // for options
+  const options = {
+    serverBrowsers(defaults) {
+      return [];
     }
   };
 
-  // Coverage reporting
-  // Coverage is enabled by passing the flag --coverage to npm test
-  var coverageFlag = process.env.npm_config_coverage;
-  var reportCoverage = process.env.TRAVIS || coverageFlag;
-  if (reportCoverage) {
-    settings.browserify.transform.push('browserify-istanbul');
-    settings.reporters.push('coverage');
-  }
-
-  if (process.env.TRAVIS) {
-    if (process.env.BROWSER_STACK_USERNAME) {
-      settings.browsers = [
-        'chrome_bs',
-        'firefox_bs',
-        'safari_bs',
-        'edge_bs',
-        'ie11_bs'
-      ];
-    } else {
-      settings.browsers = ['chrome_travis'];
-    }
-  }
-
-  config.set(settings);
-};
-
-function getCustomLaunchers(){
-  return {
-    chrome_travis: {
-      base: 'Chrome',
-      flags: ['--no-sandbox']
-    },
-
-    chrome_bs: {
-      base: 'BrowserStack',
-      browser: 'chrome',
-      os: 'Windows',
-      os_version: '10'
-    },
-
-    firefox_bs: {
-      base: 'BrowserStack',
-      browser: 'firefox',
-      os: 'Windows',
-      os_version: '10'
-    },
-
-    safari_bs: {
-      base: 'BrowserStack',
-      browser: 'safari',
-      os: 'OS X',
-      os_version: 'High Sierra'
-    },
+  config = generate(config, options);
+
+  config.frameworks.push('browserify');
+  config.plugins.push('karma-browserify');
+  config.browserify = {
```

Comment:
ideally this would all be in rollup, but the tests are heavily reliant on some things browserify does right now, so that will be a big job.